lib/Runner.js

```diff
@@ -86,7 +76,11 @@ function moduleFromFilePath(filePathArg) {
   // e.g. 'tests/All/Passing.elm' ===> ['Passing', 'All', 'tests']
   // This way, if we're given 'All.Passing' as a module name, we can also
   // flip it into ['Passing', 'All'], and see if the first N elements line up.
-  return _.compact(parsed.dir.split(path.sep).concat([basename])).reverse();
+  return parsed.dir
+    .split(path.sep)
+    .concat([basename])
+    .filter(Boolean)
```

Comment:
Out of curiousity: what does `.filter(Boolean)` do here?

Reply:
`Boolean` has the type `a -> Bool`. It turns truthy things into `true`, and falsy things into `false`. So the effect of `.filter(Boolean)` is that you only keep truthy things. Flow also has a special case for `.filter(Boolean)` that actually changes the type of the array (so you can get rid of `null`s for example: https://github.com/facebook/flow/blob/b9f3f436905537817f64e129f3428f612aba6683/lib/core.js#L225

Reply:
I switched to explicitly filtering out empty strings.